.github/workflows/R-CMD-check-hard.yaml

```diff
@@ -52,6 +52,14 @@ jobs:
             any::rmarkdown
           needs: check
 
+      - name: Install Linux dependencies
+        if: runner.os == 'Linux'
+        run: |
+          sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable
```

Comment:
On 18.04, Ubuntu only has extremely old versions of spatial libraries. By adding this ppa we can get their more modern versions, and ensure that sf uses the modern versions of these libraries by building from source below. Note that this is a pretty common recommendation -- it's on the sf README for instance: https://github.com/r-spatial/sf